src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -752,6 +779,46 @@ private static bool TryGetIntQueryParameterValue(NameValueCollection queryString
             }
         }
 
+        private async Task<HttpResponseMessage> HandleRestartInstanceRequestAsync(
+            HttpRequestMessage request,
+            string instanceId)
+        {
+            try
+            {
+                IDurableOrchestrationClient client = this.GetClient(request);
+
+                var queryNameValuePairs = request.GetQueryNameValuePairs();
+
+                string newInstanceId;
+                if (TryGetBooleanQueryParameterValue(queryNameValuePairs, RestartWithNewInstanceId, out bool restartWithNewInstanceId))
+                {
+                    newInstanceId = await client.RestartAsync(instanceId, restartWithNewInstanceId);
+                }
+                else
+                {
+                    newInstanceId = await client.RestartAsync(instanceId);
+                }
+
+                if (TryGetTimeSpanQueryParameterValue(queryNameValuePairs, TimeoutParameter, out TimeSpan? timeout)

```

Comment:
we should consider following the new logic Sebastian implemented in the StartOrchestration API for the netherite provider.

Reply:
Honestly, in general, I don't love exposing the `WaitForCompletionOrCreateCheckStatusResponseAsync` option on the HTTP webhook URIs, as in app service HTTP requests are limited to 230 seconds anyways. I can see the argument for leaving it in, but since this is a new API we don't _need_ to include it.

Reply:
I see. That makes sense. I feel like because it is in the start orchestrator method I should have it in this restart method in case a customer was using that feature in their start method. I'll add the Sebastian logic in, though.